budgetApp1/budgetApp1/SignUpViewController.swift

```diff
@@ -9,7 +9,8 @@ class SignUpViewController: UIViewController {
     @IBOutlet private weak var password: UITextField!
     
     // MARK: - Variables
-    let userViewModel = UserViewModel.shared // Use shared instance
+   
+    private lazy var userViewModel = UserViewModel(authenticationRepository: AuthenticationRepository())
     
     @IBAction func signUpClicked(_ sender: Any) {
         guard let username = username.text,
```

Comment:
don't forget to add marks for IBActions as well 💪

Reply:
Yessir!